client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/operations/UntracedHotRodOperation.java

```diff
@@ -0,0 +1,29 @@
+package org.infinispan.client.hotrod.impl.operations;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import net.jcip.annotations.Immutable;
+import org.infinispan.client.hotrod.DataFormat;
+import org.infinispan.client.hotrod.ProtocolVersion;
+import org.infinispan.client.hotrod.configuration.Configuration;
+import org.infinispan.client.hotrod.impl.protocol.Codec;
+import org.infinispan.client.hotrod.impl.transport.netty.ChannelFactory;
+
+@Immutable
+public abstract class UntracedHotRodOperation<T> extends HotRodOperation<T> {
+
+   protected UntracedHotRodOperation(short requestCode, short responseCode, Codec codec, int flags, Configuration cfg,
+                                     byte[] cacheName, AtomicInteger topologyId, ChannelFactory channelFactory,
+                                     DataFormat dataFormat) {
+      super(requestCode, responseCode, chooseCodec(codec), flags, cfg, cacheName, topologyId, channelFactory, dataFormat);
+   }
+
+   protected UntracedHotRodOperation(short requestCode, short responseCode, Codec codec, int flags, Configuration cfg,
+                                     byte[] cacheName, AtomicInteger topologyId, ChannelFactory channelFactory) {
+      super(requestCode, responseCode, chooseCodec(codec), flags, cfg, cacheName, topologyId, channelFactory);
+   }
+
+   private static Codec chooseCodec(Codec codec) {
+      return codec.supportsTracing() ? ProtocolVersion.getWithoutTracing() : codec;
+   }
+}
```

Comment:
This is a very nice solution that I think it was alternative to adding the `safePingVersionCodec` field to the `OperationsFactory`.
I like both the solutions.
Let's decide which one we want to keep it

Reply:
I did this as a way to be more explicit if the command should be traced or not. If we have more commands in the future that we do not want to trace we could just change from which class it extends. The version only using the negotiated codec is simpler though, no problems if we want to stick with that.

Reply:
It is a nice idea to mark the protocol as ready to support the tracing or not. But I think that the changes we've made with Hot Rod v.4.0 are not only for tracing, but have a more general purpose scope.
Moreover, in my opinion it looks too complex to me having a for loop to find the first safe Hot Rod codec version.
In general if something could be implemented with less code and a simpler solution, I prefer that way.